tests/test_project.py

```diff
@@ -146,6 +147,40 @@ def test_save_project_artifact(tmp_path):
     assert artifact == [0, 1, 2]
 
 
+@patch("lazyscribe.artifacts.joblib.version", side_effect=["1.2.2", "0.0.0"])
+def test_save_project_artifact_failed_validation(mock_version, tmp_path):
+    """Test saving and loading project with an artifact."""
+    location = tmp_path / "my-project"
+    location.mkdir()
+    project_location = location / "project.json"
+    today = datetime.now()
+
+    datasets = pytest.importorskip("sklearn.datasets")
+    svm = pytest.importorskip("sklearn.svm")
+
+    project = Project(fpath=project_location, author="root")
+    with project.log(name="My experiment") as exp:
+        # Fit a basic estimator
+        X, y = datasets.make_classification(n_samples=100, n_features=10)
+        estimator = svm.SVC(kernel="linear")
+        estimator.fit(X, y)
+        exp.log_artifact(name="estimator", value=estimator, handler="joblib")
+
+    project.save()
+
+    assert project_location.is_file()
+    assert (
+        location / f"my-experiment-{exp.last_updated.strftime('%Y%m%d%H%M%S')}" / "estimator.joblib"
+    ).is_file()
+
+    # Reload project and validate experiment
+    with pytest.raises(RuntimeError):
+        project2 = Project(project_location, mode="r")
+        exp2 = project2["my-experiment"]
+        model_load = exp2.load_artifact(name="estimator")

```

Comment:
Wait why does this raise an error again?

Reply:
Because I change the `package_version` that's outputted from `importlib_metadata.version` by patching that module. So there's a version mismatch